suse_migration_services/units/product_setup.py

```diff
@@ -58,29 +58,61 @@ def main():
         products_metadata = os.sep.join(
             [root_path, 'etc', 'products.d']
         )
-        baseproduct = os.sep.join(
-            [products_metadata, 'baseproduct']
-        )
-        if os.path.exists(baseproduct):
+        if os.path.exists(products_metadata):
             log.info('Creating backup of Product data')
             Command.run(
                 [
                     'rsync', '-zav', '--delete', products_metadata + os.sep,
                     '/tmp/products.d.backup/'
                 ]
             )
-            log.info('Updating Base Product to be suitable for migration')
-            xml = ElementTree()
-            xml.parse(baseproduct)
-            register_sections = xml.findall('register')
-            for register in register_sections:
-                target_sections = register.findall('target')
-                for target in target_sections:
-                    register.remove(target)
-            xml.write(
-                baseproduct, encoding="UTF-8", xml_declaration=True
-            )
+        baseproduct_filename = get_baseproduct(products_metadata)
+        if len(baseproduct_filename) == 1:
+            baseproduct = baseproduct_filename.pop()
+        else:
+            if not baseproduct_filename:
+                log.error('There is no baseproduct')
+                raise DistMigrationProductSetupException(
+                    'There is no baseproduct'
+                )
+            else:
+                log.error('There are more than 1 product without flavor tag')
+                raise DistMigrationProductSetupException(
+                    'There are more than 1 product without flavor tag'
+                )
+        log.info('Updating Base Product to be suitable for migration')
+        update_baseproduct(baseproduct)
     except Exception as issue:
         message = 'Base Product update failed with {0}'.format(issue)
         log.error(message)
         raise DistMigrationProductSetupException(message)
+
+
+def get_baseproduct(products_metadata):
+    baseproduct_filename = []
+    xml = ElementTree()
+    for product_name in os.listdir(products_metadata):
+        product = os.sep.join(
+            [products_metadata, product_name]
+        )
+        xml.parse(product)
```

Comment:
This does not account for a user having put crap into the products.d directory by accident. We should consider only .prod files, i.e. use glob.glob(products_metadata + '/*.prod') that gives us a higher chance of only getting the files we put there.

You also need to catch any exceptions the parser may throw.